### PR TITLE
Add CreateScriptInstance

### PR DESCRIPTION
**Pull request by EladNLG**
Adds a function that converts entities to SQObjects to the SquirrelManagers.

### CODE REVIEW

File: primedev/squirrel/squirrel.h [hunk 1]
```diff
@@ -447,7 +448,7 @@ inline VoidFunction SQMessageBufferPushArg(Vector3& arg) {
 // Vectors
 template <ScriptContext context>
 inline VoidFunction SQMessageBufferPushArg(SQObject* arg) {
-	return [arg]{ g_pSquirrel<context>->pushSQObject(g_pSquirrel<context>->m_pSQVM->sqvm, arg); };
+	return [arg]{ g_pSquirrel<context>->pushobject(g_pSquirrel<context>->m_pSQVM->sqvm, arg); };
```
Comment (catornot) [inline primedev/squirrel/squirrel.h:451]: `SQMessageBufferPushArg` should probably not be overloaded for `SQObject*` since some sqobjects are managed by the sqvm

**Review by catornot — APPROVED** [review]
looks good, checked the offsets in ghidra 👍 
didn't test in game.

**Comment by EladNLG** [comment]
Ok, genuinely, how can this be even tested?

File: primedev/squirrel/squirrel.h (hunk 1 above)
Comment (GeckoEidechse) [inline primedev/squirrel/squirrel.h:451]: I'm stupid, can you ELI5 this change to me? ^^"

**Comment by GeckoEidechse** [comment]
> Ok, genuinely, how can this be even tested?

Usually the approach is to have a small PoC script mod that highlights the change. I don't understand enough atm to tell you whether that would make sense in this case ^^"

File: primedev/squirrel/squirrel.h (hunk 1 above)
Comment (EladNLG) [reply primedev/squirrel/squirrel.h:451]: Somebody failed their spelling bee and the compiler for some reason didnt figure it out until I tried to call the function, and THEN it started complaining about `pushSQObject` not existing (because it is supposed to be `pushobject` instead.)

Basically, compiler is dumb.

**Comment by EladNLG** [comment]
> > Ok, genuinely, how can this be even tested?
> 
> Usually the approach is to have a small PoC script mod that highlights the change. I don't understand enough atm to tell you whether that would make sense in this case ^^"

Unfortunately, no known part of the codebase (outside of #691) uses pushobject/createscriptinstance. My guess is that #691 will test this by proxy?

**Comment by catornot** [comment]
I did test the server offsets in a plugin :p

File: primedev/squirrel/squirrel.h (hunk 1 above)
Comment (GeckoEidechse) [reply primedev/squirrel/squirrel.h:451]: Copy that, fast tracking that fix o7

**Comment by GeckoEidechse** [comment]
Maybe we just merge without further testing ig?

**Comment by catornot** [comment]
> Maybe we just merge without further testing ig?

Yeah, just merge it.

**Comment by GeckoEidechse** [comment]
Aight consider it merged. I might just gonna put a few more fixes in front to reduce the amount of cherry-picking I have to do ^^

**Comment by catornot** [comment]
Wow this got finally merged
